plugins/techdocs-node/src/stages/generate/mkdocsPatchers.ts

```diff
@@ -108,29 +112,42 @@ export const patchMkdocsYmlPreBuild = async (
   scmIntegrations: ScmIntegrationRegistry,
 ) => {
   await patchMkdocsFile(mkdocsYmlPath, logger, mkdocsYml => {
```

Comment:
Hey - I think the logic in this function can be simplified to something like this?

```typescript
export const patchMkdocsYmlPreBuild = async (
  mkdocsYmlPath: string,
  logger: Logger,
  parsedLocationAnnotation: ParsedLocationAnnotation,
  scmIntegrations: ScmIntegrationRegistry,
) => {
  await patchMkdocsFile(mkdocsYmlPath, logger, mkdocsYml => {
    if (!('repo_url' in mkdocsYml) || !('edit_uri' in mkdocsYml)) {
      // Add edit_uri and/or repo_url to mkdocs.yml if it is missing.
      // This will enable the Page edit button generated by MkDocs.
      // If the either has been set, keep the original value
      const result = getRepoUrlFromLocationAnnotation(
        parsedLocationAnnotation,
        scmIntegrations,
        mkdocsYml.docs_dir,
      );

      if (result.repo_url || result.edit_uri) {
        mkdocsYml.repo_url = mkdocsYml.repo_url || result.repo_url;
        mkdocsYml.edit_uri = mkdocsYml.edit_uri || result.edit_uri;

        logger.info(
          `Set ${JSON.stringify(
            result,
          )}. You can disable this feature by manually setting 'repo_url' or 'edit_uri' according to the MkDocs documentation at https://www.mkdocs.org/user-guide/configuration/#repo_url`,
        );
        return true;
      }
    }
    return false;
  });
};
```

The main logic change being that now if either the `repo_url` or `edit_uri` are not set we default them 

```typescript
if (!('repo_url' in mkdocsYml) || !('edit_uri' in mkdocsYml)) {
``` 

while still preserving the previous values if they were set in the mkdocs yaml 

```typescript
mkdocsYml.repo_url = mkdocsYml.repo_url || result.repo_url;
mkdocsYml.edit_uri = mkdocsYml.edit_uri || result.edit_uri;
```

I did a quick test including the new test you created `should add edit_uri to mkdocs.yml with existing repo_url` and it all seemed to pass

Reply:
@alexlorenzi looks good to me, just tested locally and updated the commits with that change

Reply:
@isabeltomb glad to see that this solved your use case! Thanks for the quick turn around 💯